private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts

```diff
@@ -791,6 +793,92 @@ export class TemplateServiceClient {
     return { success: true, result: resultTemplates };
   }
 
+  /**
+   * @public
+   * Get entry point.
+   * Returns the specified template with the data bound into the template JSON.
+   * If a version is passed in, it will update the template JSON of that version.
+   * If not version is passed, it will update the latest version.
+   * @param {string} templateId - unique template id
+   * @param {JSON} data - data json to bound with template json
+   * @param {string} version - version number, used with templateId
+   * @param {boolean} isClient - used to ignore updating the hit number on a template
+   * @returns Promise as valid json
+   */
+  public async bindData(
+    token: string,
+    templateId: string,
+    data: JSON,
+    version?: string,
+    isClient?: boolean,
+
+  ): Promise<JSONResponse<ITemplate[]>> {
+    let authCheck = this._checkAuthenticated(token);
+    if (!authCheck.success) {
+      return authCheck;
+    }
+    let authId = this.authProvider.getAuthIDFromToken(token || this.authProvider.token);
+    let userResponse = await this._getUser(authId);
+    if (!userResponse.success || !userResponse.result || userResponse.result.length === 0) {
+      return { success: false, errorMessage: userResponse.errorMessage };
+    }
+
+    let userId = userResponse.result![0]._id;
+    const templateQuery: Partial<ITemplate> = {
+      _id: templateId,
+    };
+
+    let response = await this.storageProvider.getTemplates(templateQuery, undefined, undefined);
+
+    if (!response.success || !response.result) return response;
+
+    let templates: ITemplate[] = response.result;
+
+    if (templateId && templates.length > 0) {
+      if (templates![0].owner !== userId && templates![0].isLive === false) return { success: true, result: [] };
+
+      if (isClient === undefined || isClient === false) {
+        // Update hit counter for template
+        this._incrementTemplateHits(templateId, templates![0], version);
+      }
+      sortTemplateByVersion(templates![0]);
+      if (!version) {
+        let resultTemplates: ITemplate[] = [];
+        let template: ITemplate = templates[0];
+        if (!template.instances || !template.instances[0]) {
+          return { success: false, errorMessage: "No template instances found" };
+        }
+        let boundJSON: JSON = createCard(template.instances[0].json, data);
+        template.instances![0].json = boundJSON;
+        //return only the most recent version of the template with data bound 
+        template.instances = [template.instances[0]];
+        resultTemplates.push(template);
+        return { success: true, result: resultTemplates };
+      }
+    }
+
+
+    // Filter for the latest template version (instance)
+    let resultTemplates: ITemplate[] = [];
+    for (let template of templates) {
```

Comment:
We can assume the resulting templates has only one object, we should fail if it doesn't have exactly one element since we searched by templateid